tracee-rules/engine/engine.go

```diff
@@ -47,7 +53,7 @@ func NewEngine(sigs []types.Signature, sources EventSources, output chan types.F
 	engine.logger = *log.New(logWriter, "", 0)
 	engine.inputs = sources
 	engine.output = output
-	engine.parsedEvents = parsedEvents
+	engine.config = config
```

Comment:
we should validate the configuration. what if the channel size is passed with a negative value?

Reply:
Would changing the size to a uint be fine in this case?

Reply:
Continued below